packages/components/src/InputText/InputText.tsx

```diff
@@ -50,14 +61,54 @@ function InputTextInternal(
     },
   }));
 
+  const [currentRows, setRows] = useState(initializeRows());
+
   return (
     <FormField
+      {...props}
       type={props.multiline ? "textarea" : "text"}
       ref={inputRef}
-      {...props}
+      onChange={handleChange}
+      rows={currentRows}
     />
   );
 
+  function initializeRows() {
+    if (!props.rows) return;
+    if (typeof props.rows === "number") {
+      return props.rows;
+    } else {
+      return props.rows.min;
+    }
```

Comment:
I _think_ you might be able to simplify this with:

```suggestion
    if (typeof props.rows === "object") {
      return props.rows.min;
    }
    return props.rows;

```